File: 100 challenge/day 5/large perimeter triangle/code.rb
# @param {Integer[]} nums
# @return {Integer}
def largest_perimeter(a)
  a.sort.reverse.each_cons(3).each{|(x,y,z)|
      return x+y+z if y+z > x
  }
 0
end

def largest_perimeter(a)
  z = a.sort.reverse

  (2...z.size).each do |i|
    a=z[i]
    b=z[i-1]
    c=z[i-2]
      return a+b+c if c < a+b
  end
  0
end

File: 100 challenge/day 5/large perimeter triangle/code.py
class Solution:
     def largestPerimeter(self, nums: List[int]) -> int:
        # Condition of a triangle a<(b+c) and a>=b>=c
        nums=sorted(nums,reverse=True)
        for i in range(len(nums)-2):
            a=nums[i]
            b=nums[i+1]
            c=nums[i+2]
            if a<b+c:
                return a+b+c
        return 0
        